src/Gii.php

```diff
@@ -13,31 +12,24 @@
 final class Gii implements GiiInterface
 {
     /**
-     * @param array<string, GeneratorInterface|LazyGenerator> $generators
+     * @param array<string, GeneratorInterface|GeneratorProxy> $proxies
      */
-    public function __construct(private array $generators)
+    public function __construct(private array $proxies)
     {
     }
 
     public function addGenerator(GeneratorInterface $generator): void
     {
-        $this->generators[$generator::getId()] = $generator;
+        $this->proxies[$generator::getId()] = new GeneratorProxy(fn () => $generator, $generator::class);
```

Comment:
Method gets ready-to-use generator. Seems, proxy not need here.
Or method should allow to pass array configuration of generator.

Reply:
Done